pkg/internal/controller/controller.go

```diff
@@ -228,11 +230,14 @@ func (c *Controller) reconcileHandler(obj interface{}) bool {
 		return true
 	}
 
+	log := c.Log.WithValues("name", req.Name, "namespace", req.Namespace)
+	ctx := logf.SetContext(context.Background(), log)
+
 	// RunInformersAndControllers the syncHandler, passing it the namespace/Name string of the
 	// resource to be synced.
-	if result, err := c.Do.Reconcile(req); err != nil {
+	if result, err := c.Do.Reconcile(ctx, req); err != nil {
 		c.Queue.AddRateLimited(req)
-		c.Log.Error(err, "Reconciler error", "name", req.Name, "namespace", req.Namespace)
+		log.Error(err, "Reconciler error")
```

Comment:
One thing that has long been a problem with this log line, at least if you're using klogr, is that the output shows only the basename of **this** file, `controller.go`, as the source of the log message. It does not include any provenance about the error, so you essentially see something like this:

```
controller.go:248] controller-runtime/controller "msg"="Reconciler error" "error"="oops" "controller"="kubeadmcontrolplane" "name"="bmo-control-plane" "namespace"="default"
```

There is no way to tell where the error message actually came from, as controller.go:248 refers to this file.

It would be really nice if:
- We logged the full path of the file
- We logged the full path and line where the error came from (assuming we have stack information for it)

This may not be possible in a generic way.

Reply:
Another good reason to completely remove this log line :)

Reply:
If we are not able to log the provenance of the error, 👍 to removing it.

Reply:
Or make it V=5 or 10 in case you really want to know that it's recognizing the error in this spot (for the AddRateLimited call)